services/streammanager/kafka/client/client.go

```diff
@@ -64,6 +66,33 @@ func New(network string, addresses []string, conf Config) (*Client, error) {
 		}
 	}
 
+	if conf.SSHConfig != nil {
+		signer, err := ssh.ParsePrivateKey([]byte(conf.SSHConfig.Key))
+		if err != nil {
+			return nil, fmt.Errorf("cannot parse SSH private key: %w", err)
+		}
+
+		sshConfig := &ssh.ClientConfig{User: conf.SSHConfig.User, Auth: []ssh.AuthMethod{ssh.PublicKeys(signer)}}
+		if conf.SSHConfig.AcceptAnyHostKey {
+			sshConfig.HostKeyCallback = ssh.InsecureIgnoreHostKey()
+		}
+
+		dialer.DialFunc = func(ctx context.Context, network, address string) (net.Conn, error) {
+			sshClient, err := ssh.Dial("tcp", conf.SSHConfig.Host, sshConfig)
+			if err != nil {
+				return nil, fmt.Errorf("cannot dial SSH host %q: %w", conf.SSHConfig.Host, err)
+			}
+
+			conn, err := sshClient.Dial(network, address)
+			if err != nil {
+				return nil, fmt.Errorf(
+					"cannot dial address %q over SSH (host %q): %w", address, conf.SSHConfig.Host, err,
+				)
+			}
+			return conn, nil
+		}
+	}
```

Comment:
Should we set the net Dialer with Timeout if not defined?

```suggestion
	}
	dialer.DialFunc = net.Dialer{
		Timeout: c.config.DialTimeout,
	}.DialContext
```

Reply:
I've added a timeout in the SSH config. It works as expected when dialing. Wdyt?